### PR TITLE
added mefi_amazon_affiliate_redirector.js script

### PR DESCRIPTION
**Pull request by davekaufman**
I've had this affiliate redirect script installed forever, not even sure where I got it from originally.
Per the recent discussion around [funding metafilter](https://metatalk.metafilter.com/24814/State-of-the-Site-Metafilter-financial-update-and-future-directions#1304366), I thought it was high time I added it to the repo here for sharing with others.

### CODE REVIEW

File: userscripts/mefi_amazon_affiliate_redirector.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// ==UserScript==
+// @name          Metafilter Amazon Affiliate Redirector
+// @namespace
+// @description   Plugs Amazon Links To A Designated Affiliate
+// @include       *
```
Comment (emanuelfeld) [inline userscripts/mefi_amazon_affiliate_redirector.js]: should be 
```// @include         https://*.metafilter.com/*```

**Comment by emanuelfeld** [comment]
What about just redirecting to smile if you're on an Amazon site?

**Comment by davekaufman** [comment]
but that's not the purpose of this script?  this script's purpose is to rewrite amazon links that one encounters anywhere to include the metafilter referrer tag, no matter what site.  

redirecting to smile.amazon instead of www.amazon is in fact what I have the script doing personally, but I figured that for a broader audience the redirect to smile might be unexpected.

**Comment by jcushman** [comment]
I'm a little wary that if a bunch of people used this script it would generate suspicious traffic that would risk getting Mefi's Amazon affiliate account banned. I can't find a page that weighs in one way or another on how likely that is -- closest I found is [this post](https://engagedforums.com/discussions/Amazoncom_Associates/Discussions/The_End_is_Nigh_for_me/am-associhelp/49983.1?redirCnt=1&nav=messages&mobile=y) where someone's account was banned because "You are using Content or Special Links, or otherwise linking to the Amazon Site, on or in connection with a browser plug-in, toolbar, extension, or other client-side software."

Anyway, unless there's some clear indication that's not a risk, I'm inclined to say it's not worth it? But I'm open to other thoughts.

**Comment by davekaufman** [comment]
Ok, consensus seems to be no to this script. pulling PR, no problem! 😄

**Comment by Samizdata** [comment]
Sorry.  Was at work.  And I don't do a job allowing for me to deal with this stuff.  Despite The Blue's funding woes, I could foresee this causing issues with possibly getting the affiliate link pulled so I think pulling this is not a bad idea.

**Comment by jcushman** [comment]
@davekaufman I should have added that I think this is a great impulse, and I'm definitely wondering how else we could help out the site. Maybe some UI experiments for scary things that just might work, like mixing in more social sharing buttons? Anyway, wanted to say thanks for the effort on this even though you ended up pulling it.

**Comment by emanuelfeld** [comment]
Btw, I've messaged the mods a couple of times to see if we can get the https://mefiscripts.github.io website linked somewhere in the MetaTalk sidebar. I got a positive reply quite some time back and then sent a follow up to Josh about a month ago, but haven't heard back.

Could one of you give it a shot?

I think the scripts provide added value to the site and it'd be great if we could feature them where people might see them!

**Pull request closed without merging**